### ISSUE TITLE
New option: `jsx-one-expression-per-line` `non-jsx`

### ISSUE THREAD

**Issue by reediculous456** [jsx-eslint/eslint-plugin-react#3669]
Examples of **incorrect** code for this rule:
```jsx
<Hello
  foo="bar
>Children
</Hello>
```
Examples of **correct** code for this rule:
```jsx
<Hello
  foo="bar
>
  Children
</Hello>
```

**Comment by ljharb** [comment]
See https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-one-expression-per-line.md

**Comment by reediculous456** [comment]
I don't believe this rule quite accomplishes my use case. For example, I want this to be correct:
```jsx
<>Some text is: {variable} and goes here.</>
```
but with that rule it wants to make it: 
```jsx
<>
  Some text is:
  {` `}
  {variable}
  {` `}
  and goes here.
</>
```

I'm specifically looking to prevent the case of having multiline props but not multiline children

**Comment by ljharb** [comment]
If you set `allow` to `literal`?

**Comment by reediculous456** [comment]
Ends up with the same result. I think the rule interprets putting in a variable as a non-literal causing it to force it to multiline.

The `single-child` option seems closest to what I want, but I would want it to consider` Some text is: {variable} and goes here.` as a single child and allow that to stay on one line

**Comment by ljharb** [comment]
ah, well sure, a variable isn't a literal.

hmm, that seems like a reasonable option to add - `non-jsx`, which would allow a single line child for anything that's not a literal jsx element?

**Comment by burtek** [comment]
@ljharb happy to tackle this in coming days

**Comment by ljharb** [comment]
That'd be great, thanks!

**Comment by burtek** [comment]
@ljharb So just to be sure I understand it correctly. The `allow: 'non-jsx'` would disallow:

```jsx
<Foo>
    <Bar /><Bar /><Bar />
</Foo>
```

```jsx
<Foo>
    aaa<Bar />bbb
</Foo>
```

but allow:

```jsx
<Foo>
    aaa {variable} bbb
</Foo>
```

Right?

**Comment by reediculous456** [comment]
@ljharb whats the release timeline for this project? I see there haven’t been any releases since August and there are a couple of merged things I’m waiting on

**Comment by ljharb** [comment]
@reediculous456 there's no strict timeline, but i'll likely cut a release sooner than later.

**Comment by burtek** [comment]
I hope to get #3678 in as well :)
